Rhel 8 update

pvalena opened the pull request:
Resolves bugs:

https://bugzilla.redhat.com/show_bug.cgi?id=1717323
https://bugzilla.redhat.com/show_bug.cgi?id=2098502
https://bugzilla.redhat.com/show_bug.cgi?id=1924587

rmetrich committed:
fix(dracut-shutdown): add cleanup handler on failure

It may happen that dracut-shutdown.service fails, for example on timeout
due to very low bandwidth.
In such case, for hardening purposes, a new dracut-shutdown-onfailure.service
unit doing dracut-shutdown.service cleanup needs to execute to make sure
switching root to an incomplete initramfs won't occur later.

Resolves: #1924587

dtardon committed:
fix(dracut-systemd): correct service dependencies

dracut-systemd.service is not an early boot service, therefore it should
not use DefaultDependencies=no. This also fixes the service's ordering
dependencies, as in its current state it is missing
Before=shutdown.target umount.target

Resolves: #1924587, #1717323

liutgnu committed:
feat(lvm): add new module lvmthinpool-monitor

Previously dracut didn't support the feature of lvm thinpool autoextend.

The feature is useful to cases such as kdump, when vmcore to be saved to a
lvm thin volume. The thinpool should be able to autoextend, otherwise an
IO error will be caused and leaves an incomplete vmcore.

There is lvm2-monitor.service and dmeventd avaliable, however
considering [1], it is not suitable for kdump and memory limited cases.

This patch achieves the same by parallel looping a shell function in the
background, which calls lvextend periodically. If thredshold reaches,
autoextend it, if not then nothing happens.

[1]: https://lists.fedoraproject.org/archives/list/kexec@lists.fedoraproject.org/message/YF254ZO3PJ3U56P4OKHV3JNYP2PJUMYX/

Signed-off-by: Tao Liu <ltao@redhat.com>

Resolves: #2098502

pvalena commented on modules.d/80lvmthinpool-monitor/start-thinpool-monitor.sh:
```diff
@@ -0,0 +1,41 @@
+#!/bin/sh
+
+type getarg > /dev/null 2>&1 || . /lib/dracut-lib.sh
```
@jamacku this is not helpful, as it means it will not know the variables / methods, that are used all over dracut modules.

jamacku replied:
@pvalena I'll have a look, it should be easy to fix. Thanks for info.

jamacku replied:
In upstream, they strait up disabled these checks - https://github.com/dracutdevs/dracut/blob/2334031a890a1101c78f986681236c76ba806d91/.shellcheckrc

But I think it is not the best solution. ShellCheck can source external files before linting. I'll try that and get back to you.

jamacku replied:
@pvalena Looks like the best way how to deal with is kind of error is to backport `.shellcheckrc` from upstream. There is also an option to declare those external variables at the top of the script (non-POSIX):

```shell
declare moddir initdir systemdsystemunitdir
```

There are some related issues on ShellCheck upstream:
- https://github.com/koalaman/shellcheck/issues/356
- https://github.com/koalaman/shellcheck/issues/2356

pvalena replied:
> .shellcheckrc

There's no such file in upstream.

jamacku replied:
here: https://github.com/dracutdevs/dracut/blob/master/.shellcheckrc

pvalena replied:
EDIT: I'm not sure why it didn't want to check out earlier... I'll sync the file from upstream, thanks!

pvalena replied:
Not sure what did happen earlier,

I run ` find -type f -name .shellcheckrc ` and got 0 results on `origin/master`....

jamacku reviewed (changes requested):
Commit messages are missing `(cherry picked from commit <XYZ>)` note.

pvalena commented:
> Commit messages are missing `(cherry picked from commit <XYZ>)` note.

Well, no sure how that'd help much- they're the identical upstream commits as in the PRs (which are linked in the BZ).

pvalena commented:
> > Commit messages are missing `(cherry picked from commit <XYZ>)` note.
> 
> Well, no sure how that'd help much- they're the identical upstream commits as in the PRs (which are linked in the BZ).

_Note: we've never linked commits AFAIR_

jamacku commented:
> _Note: we've never linked commits AFAIR_

Sure, I just noted, that the last commit on main has an upstream patch linked, so I thought you have forgot.

pvalena commented:
> Sure, I just noted, that the last commit on main has an upstream patch linked, so I thought you have forgot.

Oh you're right, the bot did that. I've never noticed :). Thanks for checking!

pvalena committed:
ci: check out .shellcheckrc from upstream

To avoid ShellCheck false positives.

https://github.com/redhat-plumbers/dracut-rhel8/pull/18

rhel-only

pvalena commented:
Scratch build tier1 tests: `Finished: SUCCESS`. Let's move this forward.
